book/src/development/adding_lints.md

```diff
@@ -612,6 +613,20 @@ output in the `stdout` part.
 
 [`dbg!`]: https://doc.rust-lang.org/std/macro.dbg.html
 
+## Conflicting lints
+
+There are several lints that deal with the same pattern but suggest different approaches. In other words, some lints
+may suggest modifications that go in the opposite direction to what some other lints already propose for the same
+code, creating conflicting diagnostics.
+
+When you are creating a lint that ends up in this scenario, the following tips should be encouraged to guide
+classification:
+
+* The only case where they should be in the same category is if that category is `restriction`. For example,
+`semicolon_inside_block` and `semicolon_outside_block`.
+* For all the other cases, they should be in different categories with different levels of allowance. For example,
```

Comment:
I would even go as far as saying that one of those lints should be in `restriction`. The reason for this is, that the restriction group is the only group where we don't allow/recommend to enable the entire group, but cherry pick lints out of.

Enabling the pedantic group and then getting conflicting lints also doesn't sound like a good UX to me.

Reply:
Done